phil/cnx-routes.md

```diff
@@ -0,0 +1,302 @@
+- [Book](#book)
+- Special
+
+### Resource
+
+A Resource is typically an image that is included as part of a [Page](#page). It is referenced in a [Page](#page) as `<img src="../resources/${SHA}/${OPTIONAL_FRIENDLY_NAME}">` in the HTML contents.
+
+It is identified by a hash of its contents and may contain an optional filename.
+
+A consequence of using a hash is that Resources are replaced but never changed.
+
+
+Examples:
+
+- https://cnx.org/resources/d47864c2ac77d80b1f2ff4c4c7f1b2059669e3e9
+- https://cnx.org/resources/d47864c2ac77d80b1f2ff4c4c7f1b2059669e3e9/Figure_01_00_01.jpg
+
+Resources are static and do not change. In order to replace an image, one has to edit the [Page](#page) and change the `<img src="..."/>` tag.
+
+
+### Page
+
+A Page is the content of a book. It is smaller than a Chapter, typically a Book Section.
+
+A Page is identified by a UUID, a version, and an optional [Book](#book) UUID.
+
+There are several variations of retrieving a Page but the URL format uses the following pattern:
+
+`https://archive.cnx.org/contents/{OPT_BOOK_IDENTIFIER:}{IDENTIFIER}{OPT_EXTENSION}`
+
+- `OPT_BOOK_IDENTIFIER` can be a [UUID](#uuid) or a [short UUID](#short-uuid) and may contain an optional `@{MAJOR_VERSION}.{MINOR_VERSION}` which indicates a particular version of the [Book](#book) this Page is in
+  - Why include the [Book](#book) identifier?
+    - typically numbering elements like Figures or Exercises are dependent on which Book this Page is in
+- `IDENTIFIER` can be a [UUID](#uuid) or a [short UUID](#short-uuid) and may contain an optional `@{MAJOR_VERSION}` which indicates a particular version of the Page
+- `OPT_EXTENSION` indicates which format should be returned. Valid values are `.html` or `.json`
+  - If an extension is not provided, the `Accept:` header is used to determine which format to return.
+
+
+Examples:
+
+- https://archive.cnx.org/contents/1d1fd537-77fb-4eac-8a8a-60bbaa747b6d
+- https://archive.cnx.org/contents/1d1fd537-77fb-4eac-8a8a-60bbaa747b6d.html
+- https://archive.cnx.org/contents/1d1fd537-77fb-4eac-8a8a-60bbaa747b6d.json
+- https://archive.cnx.org/contents/1d1fd537-77fb-4eac-8a8a-60bbaa747b6d@3.html
+- https://archive.cnx.org/contents/HR_VN3f7.html
+- https://archive.cnx.org/contents/HR_VN3f7.json
+- https://archive.cnx.org/contents/Ax2o07Ul:HR_VN3f7.html
+- https://archive.cnx.org/contents/Ax2o07Ul:HR_VN3f7
+
+And in the browser:
+
+- https://cnx.org/contents/1d1fd537-77fb-4eac-8a8a-60bbaa747b6d
+- https://cnx.org/contents/1d1fd537-77fb-4eac-8a8a-60bbaa747b6d@3
+- https://cnx.org/contents/HR_VN3f7
+- https://cnx.org/contents/HR_VN3f7@3
+- https://cnx.org/contents/Ax2o07Ul@9.74:HR_VN3f7@3
+- https://cnx.org/contents/031da8d3-b525-429c-80cf-6c8ed997733a@9.74:1d1fd537-77fb-4eac-8a8a-60bbaa747b6d
+
+
+### Book
+
+A Book is a tree which closely resembles the Table of Contents.
+
+- Leaf nodes **must** be [Page](#page)s
+  - These Pages end up being rendered as Preface, Chapter Sections, and Appendixes
```

Comment:
This is not the case yet on production but CTE will introduce end of book or end of chapter `composite-pages`, which are a type of page.